### PR TITLE
Fix Parser not reseting state between parse() calls causing envelopedEmail state to persist across many consecutive calls

### PR DESCRIPTION
**Pull request by LassoMike**

### CODE REVIEW

File: Parser.php [hunk 1]
```diff
@@ -82,8 +82,19 @@ class Parser
     public function __construct(PartFactory $partFactory)
     {
         $this->partFactory = $partFactory;
+        $this->reset();
```
Comment (saem) [inline Parser.php]: Yay, accidental complexity!

So we're rebooting to fix the actual problem, which is reusing the parser when it's not pure. It's also not respecting separation of concerns, we have to know to call the parse method prior to calling any other methods, hilariously this is one of the things I pointed out earlier to someone else, who thought it wasn't that big a deal. =D

In anycase, I believe what should be done is the parse method should return a ParsedMessage, which should have the stateful bits that are stuck on Parser right now, then it's nice and cleanly separated, and the resetting is implicit.

Comment (saem) [inline Parser.php]: I can talk you through what I mean if you'd like over voice.

Comment (LassoMike) [inline Parser.php]: I 100% get why that's a bad solution, I knew it was a bad solution when I wrote it. I am honestly concerned about re-engineering the parser. It was designed and built as a mutable object, and all of it's tests are based off of that. I don't want to potentially introduce many more bugs into such a heavily used feature.

Comment (trb) [inline Parser.php]: Yeah, I fucked this class up when I wrote it. There should be a part-class that extends Zends part-class and adds all the part-specific functionality we need (the workarounds for accessing headers, the getPrimaryContent() method, etc).

Also, the parts themselves should be stored in a tree-structure that would make it much easier to identify attachments, forwarded emails and enveloped emails. See the issue I've created for this: https://github.com/lassodatasystems/LassoMailParserBundle/issues/5

However, I don't think it's a good time to re-write the parser. Aside from the potential of introducing new bugs that tests don't cover, I don't think it's a good use of developer time right now.

**Comment by saem** [comment]
Seeing as we have a better fix in place, we no longer need this.

**Pull request closed without merging**